src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java

```diff
@@ -133,18 +135,32 @@ public UserInternshipDTO saveInternship(UserInternshipDTO internshipDTO) {
     }
 
     @Transactional
-    public void addUser(UUID userId, UUID internshipId) {
-        userRepository.findById(userId)
-                .orElseThrow(() -> new NotFoundException("User Not Found"));
-
-        internshipRepository.findById(internshipId)
+    public void addUser(List<UUID> userIds, UUID internshipId) {
+        Internship internship = internshipRepository.findById(internshipId)
                 .orElseThrow(() -> new NotFoundException("Internship Not Found"));
 
-        int exist = internshipRepository.checkUserAssign(userId, internshipId);
-        if (exist > 0) {
-            throw new RuntimeException("User is already assigned to Internship");
+        List<User> assignUsers = internship.getUsers();
+
+        List<User> existingUsers = userRepository.findAllById(userIds);
+
+        List<UUID> existingUsersIds = existingUsers.stream()
+                .map(User::getId)
+                .collect(Collectors.toList());
+
+        if (!userIds.containsAll(existingUsersIds)) {
+            throw new NotFoundException("Some Users Not Found");
         }
-        internshipRepository.addUserToInternship(userId, internshipId);
+
+        List<User> alreadyAssigned = assignUsers.stream()
+                .filter(existingUsers::contains)
+                .collect(Collectors.toList());
+
+        if (!alreadyAssigned.isEmpty()){
+            throw new RuntimeException(alreadyAssigned+" already exists");
+        }
+
+        assignUsers.addAll(existingUsers);
```

Comment:
Can't get this logic. You are adding ALL EXISTING users...

Reply:
EXISTING users - its users which exists in our db but not assign to internship

Reply:
ALL EXISTING - users which we found from received ids